update-gh-pages.sh

```diff
@@ -0,0 +1,34 @@
+#!/usr/bin/env bash
+#
+# This script will build the API reference docs, copy them to the gh-pages
+# branch, and tell you whether you need to commit changes.
+
+set -e -o pipefail
+
+CURRENT_BRANCH=$(git rev-parse --abbrev-ref HEAD)
+: ${CURRENT_BRANCH:=master}
```

Comment:
I would rather the script fail if ran on a branch other than `master`. What is the use-case for publishing from a different branch?

Reply:
I couldn't have tested the script itself (on the current PR branch), if the restriction to `master` had been in place. I don't see the value in _preventing_ it from any other branch.